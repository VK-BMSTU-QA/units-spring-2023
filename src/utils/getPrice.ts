import type { PriceSymbol } from '../types';

export const getPrice = (value: number, symbol: PriceSymbol = 'â‚½'): string =>
    `${value.toLocaleString('en-En')} ${symbol}`;
