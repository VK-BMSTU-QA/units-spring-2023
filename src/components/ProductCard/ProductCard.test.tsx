import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { ProductCard } from './ProductCard';
import { Product } from '../../types';
import { getPrice } from '../../utils/getPrice';

jest.mock('../../utils/getPrice', () => {
    return {
        __esModule: true,
        getPrice: jest.fn(() => '1,000 ₽'),
    };
});

describe('test ProductCard function', () => {
    afterEach(jest.clearAllMocks);

    it('renders correctly', () => {
        const product: Product = {
            id: 123,
            name: 'Product Name',
            description: 'Product Description',
            price: 1000,
            priceSymbol: '₽',
            category: 'Для дома',
            imgUrl: 'http://example.com/image.png',
        };

        const { asFragment } = render(<ProductCard {...product} />);

        expect(asFragment()).toMatchSnapshot();
        expect(getPrice).toHaveBeenCalledWith(
            product.price,
            product.priceSymbol
        );
    });

    it('renders without image', () => {
        const product: Product = {
            id: 12,
            name: 'Product Name',
            description: 'Product Description',
            price: 1000,
            priceSymbol: '₽',
            category: 'Одежда',
        };

        const { queryByAltText } = render(<ProductCard {...product} />);

        expect(queryByAltText(product.name)).not.toBeInTheDocument();
        expect(getPrice).toHaveBeenCalledWith(
            product.price,
            product.priceSymbol
        );
    });

    it('renders without price symbol', () => {
        const product: Product = {
            id: 34,
            name: 'Product Name',
            description: 'Product Description',
            price: 1000,
            category: 'Электроника',
            imgUrl: 'http://example.com/image.png',
        };

        const { getByText } = render(<ProductCard {...product} />);

        expect(getPrice).toHaveBeenCalledWith(product.price, undefined);
        expect(getByText('1,000 ₽')).toBeInTheDocument();
    });
});
