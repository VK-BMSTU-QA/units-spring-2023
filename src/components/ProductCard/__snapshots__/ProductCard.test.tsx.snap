// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ProductCard function renders correctly 1`] = `
<DocumentFragment>
  <div
    class="product-card"
  >
    <div
      class="product-card__text"
    >
      <h2
        class="product-card__name"
      >
        Product Name
      </h2>
      <div
        class="product-card__description"
      >
        Product Description
      </div>
      <h1
        class="product-card__price"
      >
        1 000 ₽
      </h1>
      <div
        class="product-card__category"
      >
        Для дома
      </div>
    </div>
    <img
      alt="Product Name"
      class="product-card__image"
      src="http://example.com/image.png"
    />
  </div>
</DocumentFragment>
`;
